wright/src/grammar/parsers/expression.rs

```diff
@@ -24,15 +24,32 @@ pub(crate) mod block;
 #[cfg(test)]
 mod expression_tests;
 
-use crate::grammar::ast::{eq::AstEq, Expression};
+use crate::grammar::ast::{
+    eq::AstEq, BinaryExpression, Block, BooleanLit, CharLit, Conditional, Expression, Identifier,
+    IndexExpression, NumLit, Parens, SelfLit, StringLit, UnaryExpression,
+};
 use crate::grammar::model::{Fragment, HasFragment};
-
+use nom::branch::alt;
+use nom::combinator::map;
 use nom::IResult;
 
 impl<'s> Expression<'s> {
     /// Parse an expression
     pub fn parse(input: Fragment<'s>) -> IResult<Fragment<'s>, Self> {
-        todo!("Expression::parse is unimplemented")
+        alt((
+            map(Conditional::parse, Expression::Conditional),
+            map(BinaryExpression::parse, Expression::BinaryExpression),
```

Comment:
Does Binary Expression's `parse` immediately call Expression's parse for it's left hand side? If so this would cause an infinite loop.

Reply:
Binary expressions parse does nothing at the moment. It cannot call `Expression::parse` for the left hand side in any case